feat: add ontology endpoint in REST API

fahad-shoaib opened the pull request:
## Description

This PR resolves #1446 by adding support to upload ontology files and refer to them in the cognee POST request.

  Implementation Details:
  - New endpoint: POST /api/v1/ontologies for ontology file upload with a simple key parameter
  that can be referenced in POST cognify requests
  - File storage: Ontology files are stored in /tmp/ontologies/{user_id}/ with metadata
  management
  - New service: OntologyService created for file management and metadata handling
  - Resolver: RDFLibOntologyResolver modified to handle file-like objects.


## Type of Change
<!-- Please check the relevant option -->
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Other (please specify):

## Screenshots/Videos (if applicable)
<!-- Add screenshots or videos to help explain your changes -->

## Pre-submission Checklist
<!-- Please check all boxes that apply before submitting your PR -->
- [X] **I have tested my changes thoroughly before submitting this PR**
- [X] **This PR contains minimal changes necessary to address the issue/feature**
- [X] My code follows the project's coding standards and style guidelines
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if applicable)
- [X] All new and existing tests pass
- [X] I have searched existing PRs to ensure this change hasn't been submitted already
- [X] I have linked any relevant issues in the description
- [X] My commits have clear and descriptive messages

## DCO Affirmation
I affirm that all code in every commit of this pull request conforms to the terms of the Topoteretes Developer Certificate of Origin.

hajdul88 commented on cognee/tests/test_ontology_endpoint.py:
```diff
@@ -0,0 +1,95 @@
+import pytest
```
I am not sure in this way these tests get triggered by the CI/CD. Please put them under tests/unit/api/ directory. Unit tests get triggered by basic_tests.yml where we run pytest on the tests/unit directory

fahad-shoaib commented on cognee/tests/test_ontology_endpoint.py:
```diff
@@ -0,0 +1,95 @@
+
+
+@patch.object(gau_mod, "get_default_user", new_callable=AsyncMock)
+def test_upload_ontology_unauthorized(mock_get_default_user, client, mock_default_user):
```
Got it. I’m extending the test with the full flow. Quick question though what’s the best way to confirm in the test that the ontology was actually attached to the cognify process? Should I check a specific flag, log entry, or graph property?

hajdul88 replied:
Cool, I think the simplest approach is to just check whether there are any nodes with ontology_valid=True in the graph. If an ontology node/subgraph gets connected, it will show up with that flag, so we can easily tell which subgraph comes from the ontology and which comes from the LLM.

fahad-shoaib commented:
Hi @hajdul88 , I've addressed your feedback by adding multifile support and improving tests - extended test_cognee_server_start.py and moved the ontology test under tests/unit/api. Ready for another review when you are.

hajdul88 commented on cognee/tests/test_cognee_server_start.py:
```diff
@@ -7,6 +7,7 @@
 from pathlib import Path
```
I've tried to add 2 ontologies locally, and now the ontology endpoint doesn't work for me. Got this error:

{
  "error": "Expecting value: line 1 column 1 (char 0)"
}

If you want to reproduce it, I added the examples/data/car_and_tech_companies.txt (so basically the text that we have in our ontology example) and tried to add our two owl examples from examples/python/ontology_input_example/. It is also not clear that in this case how the ontology name/description would work as we are adding multiple files at the same time.
